images/airflow/2.10.1/python/mwaa/celery/task_monitor.py

```diff
@@ -290,33 +290,87 @@ def _cleanup_undead_process(process_id: int):
-    sorted_filenames = sorted(signal_filenames)
-    for signal_filename in sorted_filenames:
-        # In case of issues, signals can arrive late and out of order. So, we scan all unprocessed signals in a search time range.
-        # Only a handful of signals are expected for each worker, so this repeated processing should be very light.
-        signal_file_path = os.path.join(MWAA_SIGNALS_DIRECTORY, signal_filename)
-        file_timestamp = os.path.getctime(signal_file_path)
-        if file_timestamp > signal_search_start_timestamp:
-            with open(signal_file_path, "r") as file_data:
-                try:
-                    signal_data = json.load(file_data)
-                except json.JSONDecodeError as e:
-                    logger.info(f"Error decoding file {signal_file_path}, signal will be ignored: {e}")
-                    signal_data = None
-                if signal_data and not signal_data["processed"]:
-                    return signal_file_path, signal_data
-    return None, None
+    @classmethod
+    def from_string(cls, type_string: str) -> 'SignalType':
+        """
+        Creates a SignalType object from a string.
+        :param type_string: String representation of the signal type
+        :return SignalType: New SignalType object with parsed value
+        :raises ValueError: If string is not a valid SignalType
+        """
+        return cls(type_string)
+
+class SignalData:
+    """
+    Class representing data associated with a signal.
+    """
 
+    def __init__(self):
+        """
+        Initialize a SignalData object with default values.
+        """
+        self.signalType: SignalType = SignalType.ACTIVATION
+        self.executionId = ""
+        self.createdAt = int(datetime.now(tz=tz.tzutc()).timestamp())
+        self.processed = False
 
-def _marked_signal_as_processed(signal_filepath, signal_data):
-    signal_data["processed"] = True
-    with open(signal_filepath, "w") as file_pointer:
-        json.dump(signal_data, file_pointer)
-    logger.info(f"Successfully processed signal {signal_data['executionId']}")
+    def __str__(self):
+        """
+        String representation of the SignalData object.
+        :return: String containing signal data in JSON format.
+        """
+        return json.dumps({
+            'executionId': self.executionId,
+            'signalType': self.signalType.value,
+            'createdAt': self.createdAt,
+            'processed': self.processed
+        })
+
+    def to_json(self):
+        """
+        Convert the SignalData object to a JSON object.
+        :return: JSON object containing signal data.
+        """
+        return {
+            'executionId': self.executionId,
+            'signalType': self.signalType.value,
+            'createdAt': self.createdAt,
+            'processed': self.processed
+        }
+
+    @classmethod
+    def from_json_string(cls, json_string: str) -> 'SignalData':
```

Comment:
Should this be `from_string` for consistency?

Reply:
The reason I named it this way was to basically give a hint to the user that the argument needs to be a JSON string. A non json string will result in parsing error. But I don't have a strong preference. Let me know if you want me to change it.

Reply:
I think it's probably fine.